### PR TITLE
Add textScale(r) value to Flutter context

### LINKED ISSUE #1832: Add textScale(r) value to Flutter context
### Problem Statement

Text scale changes currently only lead to breadcrumbs but especially with screenshots it would be helpful to see the current "global" text scale in Sentry.

In Flutter the text scale can be different for different widgets but in most cases, the system provided scale is enough.

### Solution Brainstorm

I would suggest something like `MediaQuery.maybeTextScalerOf(context)?.toString()` to be added to the context.
The output would be:

```dart
  @override
  String toString() => textScaleFactor == 1.0 ? 'no scaling' : 'linear (${textScaleFactor}x)';
```

This feature requires a context/widget so I would suggest to let the screenshot widget handle this.

### Are you willing to submit a PR?

None

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

Add Flutter text scale to event contexts.

## :bulb: Motivation and Context

Closes #1832

## :green_heart: How did you test it?

Tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

- [x] Handle deprecation

### CODE REVIEW

**Comment by buenaflor** [comment]
Blocked by: #1881, might need to be addressed first before we merge this

File: flutter/lib/src/event_processor/widget_event_processor.dart
Comment (kuhnroyal) [inline flutter/lib/src/event_processor/widget_event_processor.dart]: You can just use `textScaler?.toString() ?? 'no-scaling'` here and avoid the deprecation.

Comment (denrase) [reply flutter/lib/src/event_processor/widget_event_processor.dart]: Looks like `toString` is only overridden in `_LinearTextScaler` class, while `_ClampedTextScaler` does not override `toString`.

Comment (kuhnroyal) [reply flutter/lib/src/event_processor/widget_event_processor.dart]: Right... but since you can't figure out if it is clamped anyway, because of private classes, maybe just go with `textScaler?.toString() ?? 'unknown'`?

Comment (kuhnroyal) [reply flutter/lib/src/event_processor/widget_event_processor.dart]: Also I don't know in which Flutter version `TextScaler` was introduced. Need to be careful about not breaking backwards compatibility.

Comment (denrase) [reply flutter/lib/src/event_processor/widget_event_processor.dart]: Don't really know that the output of `toString` will be when it's non-null. Opted to use the `scale(fontSize)` function for now with value `1`. Does this add enough context so that it's useful? Don't really know how else we should handle non-liniear scalers.

**Review by kuhnroyal — APPROVED** [review]
I think this is good. The GlobalKey will lead to errors when someone tries to use multiple `SentryWidget`s - is this something to worry about?

**Comment by denrase** [comment]
@kuhnroyal The`SentryWidget` is not supposed to be used multiple times, so this should not be an issue as long as it is used correctly.

**Comment by buenaflor** [comment]
@denrase we should prolly document that somewhere (if not already)

File: flutter/lib/src/event_processor/widget_event_processor.dart
Comment (denrase) [reply flutter/lib/src/event_processor/widget_event_processor.dart]: @kuhnroyal Any additional thoughts?

Comment (kuhnroyal) [reply flutter/lib/src/event_processor/widget_event_processor.dart]: > Also I don't know in which Flutter version TextScaler was introduced. Need to be careful about not breaking backwards compatibility.

Still wondering about this and have not had the time to look into it yet.

Comment (kuhnroyal) [reply flutter/lib/src/event_processor/widget_event_processor.dart]: The `TextScaler` was only added in Flutter 3.16 - see https://docs.flutter.dev/release/breaking-changes/deprecate-textscalefactor?
So there is probably a switch needed, or for now just live with the deprecation and use the `textScaleFactor`.

Comment (buenaflor) [reply flutter/lib/src/event_processor/widget_event_processor.dart]: @denrase can we use the dynamic approach here that we do for other code that needs backwards compatibility?

Comment (denrase) [reply flutter/lib/src/event_processor/widget_event_processor.dart]: If we only have this static method to access the text scaler, we cannot use the dynamic approach. In that case we can only use the deprecated method available on all flutter versions we have to support.

**Review by kuhnroyal — APPROVED** [review]
I think this is the best possible way atm.

**Review by buenaflor — APPROVED** [review]

**Pull request merged**
